pkg/network/cluster_config.go

```diff
@@ -117,9 +123,128 @@ func validateClusterConfig(clusterConfig *configv1.Network, infraRes *bootstrap.
+	subnets = append(subnets, subnet{
+		name: "v4InternalMasqueradeSubnet",
+		cidr: v4InternalMasqueradeSubnet,
+	})
+	subnets = append(subnets, subnet{
+		name: "v6InternalMasqueradeSubnet",
+		cidr: v6InternalMasqueradeSubnet,
+	})
+
+	for i, subnetA := range subnets {
+		for j, subnetB := range subnets {
+			// do not compare the same elements
+			if i == j {
+				continue
+			}
+
+			_, netA, err := net.ParseCIDR(subnetA.cidr)
+			if err != nil {
+				return errors.Wrapf(err, "could not parse %s:%s", subnetA.name, subnetA.cidr)
+			}
+			_, netB, err := net.ParseCIDR(subnetB.cidr)
+			if err != nil {
+				return errors.Wrapf(err, "could not parse %s:%s", subnetB.name, subnetB.cidr)
+			}
+			if netA.Contains(netB.IP) || netB.Contains(netA.IP) {
+				return fmt.Errorf("network %s(%s) overlaps with network %s(%s)",
+					subnetA.name, subnetA.cidr, subnetB.name, subnetB.cidr)
+			}
+		}
+	}
+
+	return nil
+}
+
+func validateLiveMigration(clusterConfig *configv1.Network, infraRes *bootstrap.InfraStatus, client cnoclient.Client) error {
+	// If the migration is completed or is already progressing do not run the validation
+	if clusterConfig.Spec.NetworkType == clusterConfig.Status.NetworkType ||
+		meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, names.NetworkTypeMigrationInProgress, metav1.ConditionTrue) {
+		return nil
+	}
+	if clusterConfig.Spec.NetworkType != string(operv1.NetworkTypeOpenShiftSDN) &&
+		clusterConfig.Spec.NetworkType != string(operv1.NetworkTypeOVNKubernetes) {
+		return fmt.Errorf("network type live migration is only supported for OVNKubernetes and OpenShiftSDN CNI")
+	}
+
+	if infraRes.HostedControlPlane != nil {
+		return errors.Errorf("network type live migration is not supported on HyperShift clusters")
+	}
+
+	operNet := &operv1.Network{}
+	err := client.Default().CRClient().Get(context.TODO(), types.NamespacedName{Name: names.CLUSTER_CONFIG}, operNet)
+	if err != nil {
+		return errors.Errorf("error getting network configuration: %v", err)
+	}
+
+	// Status contains the CNI we are migrating from
+	if clusterConfig.Status.NetworkType == string(operv1.NetworkTypeOpenShiftSDN) {
+		if operNet.Spec.DefaultNetwork.OpenShiftSDNConfig != nil &&
+			operNet.Spec.DefaultNetwork.OpenShiftSDNConfig.Mode == operv1.SDNModeMultitenant {
+			return errors.Errorf("network type live migration is not supported on SDN Multitenant clusters")
+		}
+
+		if err := validateOVNKubernetesCIDROverlap(clusterConfig, operNet); err != nil {
+			return err
+		}
+
+		// Detect pods with pod.network.openshift.io/assign-macvlan (Used by egress router)
+		macVlanPods, err := cnoclient.ListAllPodsWithAnnotationKey(context.TODO(), client, v1.AssignMacvlanAnnotation)
+		if err != nil {
+			return errors.Wrapf(err, "error listing macvlan pods")
+		}
+		if len(macVlanPods) != 0 {
+			return fmt.Errorf("network type live migration is not supported for pods with %q annotation", v1.AssignMacvlanAnnotation)
```

Comment:
Shall we just tell the user to remove the egress router configuration before starting the live migration?

Reply:
Does this work for you?
`network type live migration is not supported for pods with %q annotation. Please remove all egress router pods`

Reply:
Done.